File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_positive_float(void);

int main(void)
{
    float dollars = get_positive_float();



    int cents = round(dollars * 100);

    int quarters = floor(cents / 25);




    int rcentsq = cents % 25;

    int dimes = floor(rcentsq / 10);



    int rcentsd = rcentsq % 10;

    int nickels = floor(rcentsd / 5);





    int rcentsn = rcentsq % 5;

    int pennies = floor(rcentsn / 1);




    int totcoins = quarters + dimes + nickels + pennies;

    printf("%i\n", totcoins);




}


// Prompt user for positive integer
float get_positive_float(void)
{
    float n;
    int p;
    do
    {
        n = get_float("Cash Input ");
        p = round(n * 100);
    }
    
    while (p < 1);
    
    return n;
}

File: pset6/cash/cash.py
from cs50 import get_float
from math import floor


def main():

    dollars = get_positive_float()

    cents = round(dollars * 100)
    quarters = floor(cents / 25)

    rcentsq = cents % 25
    dimes = floor(rcentsq / 10)

    rcentsd = rcentsq % 10
    nickels = floor(rcentsd / 5)

    rcentsn = rcentsq % 5
    pennies = floor(rcentsn / 1)

    totcoins = quarters + dimes + nickels + pennies

    print(totcoins)


def get_positive_float():
    while True:
        n = get_float("Positive Float:")

        if n > 0:
            break
    return n


main()